daemon/protocol/service.go

```diff
@@ -121,6 +121,12 @@ func (s *protocolServiceServer) InstallDriver(ctx xcontext.Context, req *Install
 
 	if ErrAlreadyInstalled.Is(err) {
 		return nil, status.New(codes.AlreadyExists, err.Error()).Err()
+	} else if errs, ok := err.(errcode.Errors); ok { //docker err codes analysis
+		for _, erro := range errs {
+			if strings.HasPrefix(erro.Error(), "unauthorized:") {
```

Comment:
This should type-assert an error instead of checking the message text

Reply:
That would be lovely, but from the experiments that I did - it does not seem possible with their usage of [errcode.Errors](https://github.com/docker/distribution/blob/master/registry/api/errcode/errors.go#L176): al `reflect.TypeOf(erro)` returns the same type.

Or am I missing something here?

Reply:
This is the trouble with implementing `error` on a raw slice. You could probably give it a method to walk through the slice and pull out the first coded value—but in the end the problem is that we're not using an error as a value but as a container. 

It'd be nice to have a test for this, though.

Reply:
@bzz Getting error code is what I assumed. I think it will be better than relying on a specific error prefix, since the error codes should be stable.